FHEM/14_SD_UT.pm

```diff
@@ -1526,10 +1526,11 @@ sub SD_UT_Parse {
 
     ### Manax MX-RCS250 | mumbi [P90] ###
     if (!$def && $protocol == 90) {
+      my $button = $models{RC_10}{buttons}{substr($bitData,20,3)};
+      if (!defined $button) {return ''}
```

Comment:
Damit wird die Verarbeitung vollständig abgebrochen.
Dispatch wird hier ebenfalls aufhören zu versuchen ein anderes Modul zu finden, welches die Nachricht verarbeiten kann. Wenn das so gewollt ist, dann kann es so bleiben. :)

Reply:
Im Normalfall war es so gedacht.
Oder spricht was dagegen @elektron-bbs ?

Reply:
Für den Fall, dass es so gewünscht ist (ohne Logmeldung etc., dann sollte das return an den Anfang gestellt werden.

```suggestion
     return ''  if (!defined $button) 
```

Achso, hast Du zufällig so eine Nachricht in dem der button nicht enthalten ist?

Reply:
Ich habe die Diskussion im Forum mitverfolgt und glaube aber nicht, das ein `return ''` die weitere Verarbeitung komplett abbricht.

```
2022.01.16 16:42:15 4: sduino_dummy: get rawmsg: MN;D=5100C6BF107F1FF8BBFFFFFFEA22;R=14;
2022.01.16 16:42:15 4: sduino_dummy: Parse_MN, 2-FSK Protocol id 100 Lacrosse mode 1 msg 5100C6BF107F1FF8BBFFFFFFEA22 not match (?^:^9)
2022.01.16 16:42:15 4: sduino_dummy: Parse_MN, 2-FSK Protocol id 103 Lacrosse mode 2 msg 5100C6BF107F1FF8BBFFFFFFEA22 not match (?^:^9)
2022.01.16 16:42:15 4: sduino_dummy: Parse_MN, Found 2-FSK Protocol id 107 -> WH51 433.92 MHz with match (?^:^51)
2022.01.16 16:42:15 4: sduino_dummy: SD_WS_Parse protocol 107, rawData 5100C6BF107F1FF8BBFFFFFFEA22
2022.01.16 16:42:15 3: sduino_dummy: SD_WS_107 Parse msg 5100C6BF107F1FF8BBFFFFFFEA22 - ERROR CRC8
2022.01.16 16:42:15 4: sduino_dummy: SD_WS_Parse 5100C6BF107F1FF8BBFFFFFFEA22 protocolid 107 (WH51, DP100, MISOL/1) - ERROR CRC
2022.01.16 16:42:15 4: sduino_dummy: Parse_MN, Found 2-FSK Protocol id 107.1 -> WH51 868.35 MHz with match (?^:^51)
2022.01.16 16:42:15 4: sduino_dummy: Dispatch, W107#5100C6BF107F1FF8BBFFFFFFEA22, Dropped due to short time or equal msg
2022.01.16 16:42:15 4: sduino_dummy: Parse_MN, Error! id 108 msg=5100C6BF107F1FF8BBFFFFFFEA22, message is to short
2022.01.16 16:42:15 4: sduino_dummy: Parse_MN, Error! id 112 msg=5100C6BF107F1FF8BBFFFFFFEA22, message is to long
2022.01.16 16:42:15 4: sduino_dummy: Parse_MN, Error! id 115 msg=5100C6BF107F1FF8BBFFFFFFEA22, message is to short
2022.01.16 16:42:15 4: sduino_dummy: Parse_MN, 2-FSK Protocol id 116 WH57 msg 5100C6BF107F1FF8BBFFFFFFEA22 not match (?^:^57)
2022.01.16 16:42:15 4: sduino_dummy: Parse_MN, 2-FSK Protocol id 116.1 WH57 msg 5100C6BF107F1FF8BBFFFFFFEA22 not match (?^:^57)
2022.01.16 16:42:15 4: sduino_dummy: Parse_MN, Error! id 117 msg=5100C6BF107F1FF8BBFFFFFFEA22, message is to short
2022.01.16 16:42:15 4: sduino_dummy: Parse_MN, Found 2-FSK Protocol id 994 -> LaCrosse mode 4
```

Nach dem `ERROR CRC` der hervorgerufen wird durch diese Zeilen:

```
    my $retcrc=$decodingSubs{$protocol}{crcok}->( $rawData,$bitData );
    if (!$retcrc) {
      Log3 $iohash, 4, "$name: SD_WS_Parse $rawData protocolid $protocol ($SensorTyp) - ERROR CRC";
      return "";
    }
```

wird munter versucht, die Nachricht weiter zu verarbeiten.
Durch ein  `return ""` wird nur die weitere Verarbeitung des einen Dispatchs abgebrochen.

Reply:
@elektron-bbs 
Ja das ist richtig. Dispatch von FHEM wird beendet (suchen eines weiteren Moduls). Wenn das 00_SIGNALduino einen weiteren Dispatch aufgrund eines weiteren Protokolls, auslöst dann wird dieser wieder an Dispatch von FHEM übergeben.

Reply:
Ich denke wir können das so belassen.
Wie ist eure finale Meinung?

Reply:
@HomeAutoUser 

Prinzipiell schon. Offen wäre noch die Frage nach einer Referenznachricht :)

Reply:
Ich habe keine RAWMSG, aber vielleicht kannst du dir aus den Warnungen eine DMSG basteln:

```
2021.11.01 07:25:15 1: PERL WARNING: Use of uninitialized value $button in concatenation (.) or string at ./FHEM/14_SD_UT.pm line 1531.
2021.11.01 07:25:15 1: PERL WARNING: Use of uninitialized value $button in string ne at ./FHEM/14_SD_UT.pm line 1534.
2021.11.01 07:25:15 1: PERL WARNING: Use of uninitialized value $button in concatenation (.) or string at ./FHEM/14_SD_UT.pm line 1559.
2021.11.01 07:25:15 1: sduinoIP: SD_UT_Parse UNDEFINED sensor unknown detected, protocol 90, data FFCE32B4E, code FFCE

2021.12.15 15:50:01 1: PERL WARNING: Use of uninitialized value $button in concatenation (.) or string at ./FHEM/14_SD_UT.pm line 1531.
2021.12.15 15:50:01 1: PERL WARNING: Use of uninitialized value $button in string ne at ./FHEM/14_SD_UT.pm line 1534.
2021.12.15 15:50:01 1: PERL WARNING: Use of uninitialized value $button in concatenation (.) or string at ./FHEM/14_SD_UT.pm line 1559.
2021.12.15 15:50:01 1: sduinoIP: SD_UT_Parse UNDEFINED sensor unknown detected, protocol 90, data AC2E43C61, code AC2E
```

Reply:
Das wäre dann P90#AC2E43C61 oder?

Reply:
Müsste passen...